### PR TITLE
[MM-53711] Set save button opacity for editing a post when disabled

### PR DESCRIPTION
**Pull request by cedricongjh**
#### Summary

The save button when editing posts appeared clickable when the post was too long, this PR sets the opacity of the save button such that the button would appear unclickable to the user.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/24617

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: 
Macbook Pro, 2020 (Intel)
MacOs 14.1.1

#### Screenshots

image

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: app/screens/edit_post/edit_post.tsx
Comment (enahum) [inline app/screens/edit_post/edit_post.tsx:9]:
```suggestion
import {changeOpacity} from '@utils/theme';
```

And sort

Comment (cedricongjh) [reply app/screens/edit_post/edit_post.tsx]: changed!

**Review by enahum — APPROVED** [review]

**Review by larkox — APPROVED** [review]

**Comment by lindalumitchell** [comment]
I'm checking with the Delivery team about the build failures.

**Comment by mvitale1989** [comment]
I can confirm that at the moment, the `Build Apps for PR` label cannot be used from PRs originating from forks. I opened a ticket to support that in the future.

To unblock QA I have pushed the source branch to this repo, and used it to create a [mirror this PR](https://github.com/mattermost/mattermost-mobile/pull/7789) and produce the artifacts from there (that mirror PR may be closed and its branch deleted, after these changes are merged). You can use this method until the long-term solution is implemented

**Review by lindalumitchell — APPROVED** [review]
Verified on iOS 17.2.1 (iPhone 13 Pro Max) and Android 14 (Pixel 5a) that the Save button is disabled and appears disabled when a post is edited to above the character limit. Save button returns to enabled (and appearing enabled) when enough characters are deleted to put the edit below the character limit. Also double-checked that the functionality works for the original post field.

Looks great, thanks @cedricongjh ! And thanks Mario for mirroring the PR to get those mobile builds built. :)

**Pull request merged**
